architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,641 @@
+
+The following statements are all accurate:
+
+- The `System.Resources/resourceProviders` type is the parent of `System.Resources/resourceProviders/resourceTypes`. 
+- The `System.Resources/resourceProviders/resourceTypes` type is the child of `System.Resources/resourceProviders`.
+- The `System.Resources/resourceProviders/resourceTypes` type is nested within `System.Resources/resourceProviders`.
+- The `System.Resources/resourceProviders` type is a top-level resource type.
+
+Any resource type with two segments (separated by `/`) is a top-level resource type. Any resource type with 3 or more segments (separate by `/`) is a nested resource type.
+
+##### Structure
+
+The resource ids (and URLs) are written like the following:
+
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders` - List all resource providers.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example` - Get the Contoso.Example resource provider.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example/resourceTypes` - List all resource types inside the `Contoso.Example` resource provider/namespace.
+- `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example/resourceTypes/contosoBuses` - Gets the `contosoBuses` resource type inside the `Contoso.Example` resource provider/namespace.
+
+---
+**Note: These ids refer to the type definition of the `Contoso.Example/contosoBuses` resource type . The ids of `Contoso.Example/contosoBuses` resources are structured like our existing resource types.**
+
+- `/planes/radius/{plane}/resourceGroups/{resourceGroup}/providers/Contoso.Example/contosoBuses` - Lists the `Contoso.Example/contosoBuses` resources inside the `/planes/radius/{plane}/resourceGroups/{resourceGroup}` scope.
+
+---
+
+Adding additional levels of nesting is allowed. eg: `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example/resourceTypes/contosoBuses/apiVersion/2024-01-01`. ARM defines an artificial limit of 3 levels of children for simplicity.
+
+
+##### Semantics
+
+Nesting a resource type also nests the lifecycle of the resource. Using our examples above, the `/planes/radius/{plane}/providers/System.Resources/resourceProviders/resourceTypes/Contoso.Example/contosoBuses` type cannot exist unless `/planes/radius/{plane}/providers/System.Resources/resourceProviders/Contoso.Example` already exists.
+
+There are a few simple rules for this:
+
+- For operations on parent resources...
+  - A `DELETE` operation on a parent resource deletes all children.
+
+- For operations on child resources...
+  - A *`CRUL`* operation (anything but delete) on a child resource must return a `404` when the parent does not exist.
+  - A `DELETE` operation on a child resource never returns a `404`, because `DELETE` operations never return `404` in our API.
```

Comment:
Would it return `Deleted` even if the resource doesn't exist?

Reply:
What is deleted?

Reply:
Resolving. If you still have questions please clarify.